packages/serve/src/lib/middleware/auth/authRouteMiddleware.ts

```diff
@@ -0,0 +1,112 @@
+import { isEmpty } from 'lodash';
+import { inject, multiInject } from 'inversify';
+import { TYPES as CORE_TYPES, VulcanInternalExtension } from '@vulcan-sql/core';
+import {
+  Next,
+  KoaContext,
+  BuiltInMiddleware,
+  BaseAuthenticator,
+  AuthOptions,
+} from '@vulcan-sql/serve/models';
+import { TYPES } from '@vulcan-sql/serve/containers';
+import * as Router from 'koa-router';
+
+type AuthenticatorMap = {
+  [name: string]: BaseAuthenticator<any>;
+};
+
+/** The auth route middleware used to auth identity.
+ *  It seek the 'auth' module name to match data through built-in and customized authenticator by BaseAuthenticator
+ * */
+@VulcanInternalExtension('auth')
+export class AuthRouteMiddleware extends BuiltInMiddleware<AuthOptions> {
+  private options = (this.getOptions() as AuthOptions) || {};
+  private authenticators: AuthenticatorMap;
+  private router = new Router();
+
+  constructor(
+    @inject(CORE_TYPES.ExtensionConfig) config: any,
+    @inject(CORE_TYPES.ExtensionName) name: string,
+    @multiInject(TYPES.Extension_Authenticator)
+    authenticators: BaseAuthenticator<any>[]
+  ) {
+    super(config, name);
+
+    this.authenticators = authenticators.reduce<AuthenticatorMap>(
+      (prev, authenticator) => {
+        prev[authenticator.getExtensionId()!] = authenticator;
+        return prev;
+      },
+      {}
+    );
+  }
+  public override async onActivate() {
+    const authIds = Object.keys(this.authenticators);
+
+    // check setup auth type in options also valid in authenticators
+    Object.keys(this.options).map((type) => {
+      if (!authIds.includes(type))
+        throw new Error(
+          `The auth type "${type}" in options not supported, authenticator only supported ${authIds}.`
+        );
+    });
+
+    for (const authId of authIds) {
+      const authenticator = this.authenticators[authId];
+      if (authenticator.activate) await authenticator.activate();
+    }
+    // setup route
+    this.setAuthRoute();
+  }
+
+  public async handle(context: KoaContext, next: Next) {
+    // return to stop the middleware, if disabled
+    if (!this.enabled) return next();
+
+    if (isEmpty(this.options))
+      throw new Error(
+        'please set at least one auth type and user credential when you enable the "auth" options.'
+      );
```

Comment:
Can we check the option at `onActivate` function too?
```ts
if (this.enable && isEmpty(xxxxx
```

Reply:
Thanks for @oscar60310 suggesting. it has been fixed.